### ISSUE TITLE
Misc journald tests fail when run as unprivileged user in CentOS 7

### ISSUE THREAD

**Issue by Ziris85** [fail2ban/fail2ban#2100]
### Environment:

- Fail2Ban version (including any possible distribution suffixes):
- OS, including release name/version: `0.10.2`
- [ ] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
On CentOS 7, when you run tests (either with `pythond setup.py test` or `./fail2ban-testcases-all`), the following tests fail when run as an unprivileged user:
```
======================================================================
ERROR: testJournalPathArg (fail2ban.tests.filtertestcase.MonitorJournalFailures<FilterSystemd>(monitorjournalfailures_FilterSystemd))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./fail2ban/tests/filtertestcase.py", line 1191, in testJournalPathArg
    self._initFilter(journalpath=jrnlpath)
  File "./fail2ban/tests/filtertestcase.py", line 1158, in _initFilter
    self.filter = Filter_(self.jail, **kwargs)
  File "./fail2ban/server/filtersystemd.py", line 62, in __init__
    self.__journal = journal.Reader(**jrnlargs)
  File "/usr/lib64/python2.7/site-packages/systemd/journal.py", line 153, in __init__
    super(Reader, self).__init__(flags, path, files)
OSError: [Errno 2] No such file or directory: ''

======================================================================
FAIL: test_WrongChar (fail2ban.tests.filtertestcase.MonitorJournalFailures<FilterSystemd>(monitorjournalfailures_FilterSystemd))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./fail2ban/tests/filtertestcase.py", line 1300, in test_WrongChar
    self.assertTrue(self.isFilled(10))
AssertionError: False is not true

======================================================================
FAIL: test_delJournalMatch (fail2ban.tests.filtertestcase.MonitorJournalFailures<FilterSystemd>(monitorjournalfailures_FilterSystemd))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./fail2ban/tests/filtertestcase.py", line 1269, in test_delJournalMatch
    self.assert_correct_ban("193.168.0.128", 3)
  File "./fail2ban/tests/filtertestcase.py", line 1205, in assert_correct_ban
    self.assertTrue(self.waitFailTotal(test_attempts, 10)) # give Filter a chance to react
AssertionError: False is not true

======================================================================
FAIL: test_grow_file (fail2ban.tests.filtertestcase.MonitorJournalFailures<FilterSystemd>(monitorjournalfailures_FilterSystemd))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./fail2ban/tests/filtertestcase.py", line 1217, in test_grow_file
    self._test_grow_file()
  File "./fail2ban/tests/filtertestcase.py", line 1245, in _test_grow_file
    self.assertTrue(self.isFilled(10))
AssertionError: False is not true
```
I was able to trace the trouble to the user not having access to the systemd journal (which appears to be default behavior for unprivileged users on CentOS 7). Once I added the user to the `systemd-journal` group, the above tests no longer failed.

#### Steps to reproduce
Run either  `python setup.py test` or `./fail2ban-testcases-all` as an unprivileged user on CentOS 7.
#### Expected behavior
All tests pass, are gracefully skipped, or warn when they can't run.
#### Observed behavior
Tests fail due to insufficient privileges.
#### Any additional information
Given the nature of the software, needing access to the systemd journal makes sense. Would be nice if the tests would check if it had systemd journal access and threw an error about that before trying to run the tests.

**Comment by Ziris85** [comment]
For now, I added these few things in my spec file to work around this. First in `%pre` section:
```
%if 0%{?rhel} >= 7
        %if "%(groups|grep -o systemd-journal || whoami|grep root)"
                %global dotest 1
        %else
                %{warn:"User needs to be in 'systemd-journal' group (or be root user) for tests to complete! Tests will be skipped."}
        %endif
%endif
```
Then in `%build` section:

    %{?dotest:python setup.py test}

**Comment by sebres** [comment]
fixed in 50d7c649ba63b5ec978461f82649dae8f4e38f32
